smartlearner/optimizers/adagrad.py

```diff
@@ -0,0 +1,55 @@
+from collections import OrderedDict
+import theano.tensor as T
+
+from . import SGD
+from ..utils import sharedX
+
+
+class ADAGRAD(SGD):
+    """
+    Implements the ADAGRAD optimizer.
+
+    Reference
+    ---------
+    Duchi, J., Hazan, E., & Singer, Y. (2010).
+    Adaptive subgradient methods for online learning and stochastic optimization.
+    Journal of Machine Learning
+    """
```

Comment:
I agree, but it's explicitly mentioned in the spec of Numpy doc. 

On August 15, 2015 10:46:56 AM EDT, "Marc-Alexandre Côté" <notifications@github.com> wrote:
>> +import theano.tensor as T
>> +
>> +from . import SGD
>> +from ..utils import sharedX
>> +
>> +
>> +class ADAGRAD(SGD):
>> +    """
>> +    Implements the ADAGRAD optimizer.
>> +
>> +    Reference
>> +    ---------
>> +    Duchi, J., Hazan, E., & Singer, Y. (2010).
>> +    Adaptive subgradient methods for online learning and stochastic
>optimization.
>> +    Journal of Machine Learning
>> +    """
>
>Do you really want to add an empty line before every closing """ ? It
>seems to me a bit overkill and it is clear enough without it (anyway
>when generating the documentation I pretty sure the `__docstring__` is
>trimmed).
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/SMART-Lab/smartlearner/pull/31/files#r37136420

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.